### PR TITLE
linux: BPF improvements

### PR DESCRIPTION
**Pull request by vavrusa**
Added:
* `strflag/multiflag` support for constants
* returned `fd` is now wrapped in `t.fd`
* new constants and support for attaching BPF programs to sockets and `SO_REUSEPORT`

### CODE REVIEW

File: syscall/linux/constants.lua [hunk 1]
```diff
@@ -455,9 +455,19 @@ c.SO = strflag(arch.SO or {
...
+  MAX_PACING_RATE    = 47,
+  BPF_EXTENSIONS     = 48,
+  INCOMING_CPU       = 49,
+  ATTACH_BPF         = 50,
+  ATTACH_REUSEPORT_CBPF = 51,
+  ATTACH_REUSEPORT_EBPF = 52,
 })
 
 c.SO.GET_FILTER = c.SO.ATTACH_FILTER
+c.SO.DETACH_BPF = c.SO.SO_DETACH_FILTER
```
Comment (justincormack) [inline syscall/linux/constants.lua]: is that right? should that be `c.SO.DETACH_FILTER`?

Comment (vavrusa) [inline syscall/linux/constants.lua]: It's an alias for the same option http://lxr.free-electrons.com/source/include/uapi/asm-generic/socket.h?v=4.0#L88

Comment (justincormack) [inline syscall/linux/constants.lua]: Yes but its `c.SO.DETACH_FILTER` not `c.SO.SO_DETACH_FILTER`...

**Comment by vavrusa** [comment]
Ah right, stupid me. Fixed!

**Pull request merged**